modmesh/app/euler1d.py

```diff
@@ -132,13 +132,40 @@ class SolverConfig():
           table.
         - :meth:`columnCount()`: Get the number of columns in the
           configuration table.
-        - :meth:`get_var(key_row, key_col)`: Get the value of a configuration
-          variable based on its key.
     """
-    def __init__(self, input_data):
+    class Accessor:
```

Comment:
Why do you use an inner class?  It's less testable.  What about moving it outside in the module?

Reply:
My original idea is to achieve `__getitem__` chaining by returning an `Accessor` instance and `Accessor` is specialize for accessing `GUIConfig` data. That's why I put this class inside `GUIConfig`.

If we move `Accessor` outside, it would imply that `Accessor` needs to become a more generic class.
I will try to re-design `Accessor` class.

Reply:
Being specific or general has nothing to do with making it an inner class or a normal (in-module) class.  When making a help class for a specific use inside another class, a common style is to mark it as a private class like:

```python
class _Accessor:
    ....
class Consumer:
    def func(self):
        a = _Accessor(...)
        a.helper_function()
```